jetson/filter/src/__main__.py

```diff
@@ -136,9 +141,24 @@ def __init__(self):
         self.lcm.subscribe("/imu_data", self._imuCallback)
         self.lcm.subscribe("/nav_status", self._navStatusCallback)
         self.lcm.subscribe("/drive_vel_data", self._driveVelDataCallback)
+        self.lcm.subscribe("/target_bearing", self._targetBearingCallback)
+        self.lcm.subscribe("/drive_vel_cmd", self._driveVelCmdCallback)
+
+        # Initializations for magnetometer bearing correction
+        self.last_bearing_correction = time.time()
+        self.bearing_offset = 0
+
+        # TODO: are these inits going to cause incorrect behavior?
+        self.target_bearing = 0
+        self.prev_target_bearing = 0
```

Comment:
this is troublesome. Not sure if u can just set this to 0.

Reply:
I actually think this should be fine. Since we have to be driving in order to correct the bearing, and I assume we will always have a target bearing if we are driving somewhere, then these initial values should never actually be used to check the correction condition. ig its possible that the very first iteration of nav could cause trouble, like if weve only received a single target bearing so the previous is still the default zero, but even then I doubt that all the other conditions would pass on the very first iteration of nav driving. We'll find out for sure in testing ig. Also FYI this is currently not even using the previous target bearing or rate of target bearing, we're gonna test it without that condition check first.

Reply:
ok, as long as we are sure that this won't get triggered on the first iter thats fine, i guess we'd just add a flag for first iteration if it did. Like you said - should be obvious.

Reply:
Usually prefer to use `None` in Python when a variable has "no data" and then handle it appropriately later with if statements

Reply:
yeah I'll do that, I was being lazy